src/js/modules/sections/selectors.js

```diff
@@ -1,108 +1,77 @@
-import { createSelector } from 'reselect';
-import { objectFilter } from '../../utils.js';
+import { createSelector } from "reselect";
+import { objectFilter } from "../../utils.js";
 
 Object.filter = objectFilter;
 
-const getSections = (state) => state.sections.sections;
-const getSectionByProps = (state, props) => props.section;
-const getSubsectionsByProps = (state, props) => props.subsections;
+export const getSections = (state) => state.sections.sections;
+export const getSectionFromProps = (state, props) => props.section;
 
 /**
- * Returns an array of slugs for a section and its subsection.
+ * The selector returns a filtered sections object which contains direct and
+ *   indirect children of a target section.
  */
-export const getSectionAndSubsectionSlugs = createSelector(
-  [ getSectionByProps, getSubsectionsByProps ],
-  (section, subsections) => {
-    return [ section.slug, ...Object.keys(subsections) ];
-  }
-);
+const getSubsectionsInSectionTree = (sections, targetSection) => {
+  return Object.filter(sections, sectionTraversingUpwards => {
```

Comment:
If you look at the code, you'll see why I named it that. I thought it described the behavior well, but I admit that it doesn't describe the subsection well. Do you have a better name?